Compilation error because of EventAssertion

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

testsuite

### Describe the bug

Some simultaneous merges made main not compilable. PR https://github.com/keycloak/keycloak/pull/48752 was merged when other pulls were also adding some new tests that used the old event clases. The result is errors at compilation:

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:testCompile (default-testCompile) on project integration-arquillian-tests-base: Compilation failure: Compilation failure: 
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionResetPasswordTest.java:[518,23] cannot find symbol
[ERROR]   symbol:   method expect(org.keycloak.events.EventType)
[ERROR]   location: variable events of type org.keycloak.testsuite.AssertEvents
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionResetPasswordTest.java:[525,23] cannot find symbol
[ERROR]   symbol:   method expectLogin()
[ERROR]   location: variable events of type org.keycloak.testsuite.AssertEvents
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionResetPasswordTest.java:[525,55] cannot find symbol
[ERROR]   symbol:   method isUUID()
[ERROR]   location: class org.keycloak.testsuite.AssertEvents
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionResetPasswordTest.java:[534,23] cannot find symbol
[ERROR]   symbol:   method expect(org.keycloak.events.EventType)
[ERROR]   location: variable events of type org.keycloak.testsuite.AssertEvents
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/UserInfoTest.java:[748,19] cannot find symbol
[ERROR]   symbol:   method expect(org.keycloak.events.EventType)
[ERROR]   location: variable events of type org.keycloak.testsuite.AssertEvents
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/UserInfoTest.java:[750,27] cannot find symbol
[ERROR]   symbol:   variable Matchers
[ERROR]   location: class org.keycloak.testsuite.oidc.UserInfoTest
[ERROR] /home/rmartinc/keycloak/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/UserInfoTest.java:[751,30] cannot find symbol
[ERROR]   symbol:   variable Matchers
[ERROR]   location: class org.keycloak.testsuite.oidc.UserInfoTest
[ERROR] -> [Help 1]
```

### Version

999.0.0-SNAPSHOT

### Regression

- [ ] The issue is a regression

### Expected behavior

Compilation should be OK.

### Actual behavior

Compilation errors.

### How to Reproduce?

just compile the main branch: `mvn clean install -DskipTests`



### Anything else?

_No response_

--- a/.github/scripts/run-fips-it.sh
+++ b/.github/scripts/run-fips-it.sh
@@ -1,8 +1,10 @@
 #!/usr/bin/env bash
 set -x
 
+JAVA_VERSION=21
+
 rm -f /etc/system-fips
-dnf install -y java-25-openjdk-devel
+dnf install -y "java-${JAVA_VERSION}-openjdk-devel"
 fips-mode-setup --enable --no-bootcfg
 fips-mode-setup --check | grep "FIPS mode is enabled."
 if [ $? -ne 0 ]; then
@@ -21,7 +23,7 @@ if [ $? -ne 0 ]; then
   exit 1
 fi
 echo "Tests: $TESTS"
-export JAVA_HOME=/etc/alternatives/java_sdk_25
+export JAVA_HOME="/etc/alternatives/java_sdk_${JAVA_VERSION}"
 set -o pipefail
 
 # Build adapter distributions
@@ -43,10 +45,10 @@ if [ $? -ne 0 ]; then
 fi
 
 # Profile app-server-wildfly needs to be explicitly set for FIPS tests
-./mvnw test -Dsurefire.rerunFailingTestsCount=$SUREFIRE_RERUN_FAILING_COUNT -nsu -B -Pauth-server-quarkus,auth-server-fips140-2,app-server-wildfly -Dredhat.crypto-policies=false $STRICT_OPTIONS -Dtest=$TESTS -pl testsuite/integration-arquillian/tests/base 2>&1 | misc/log/trimmer.sh
+./mvnw test -Dsurefire.rerunFailingTestsCount=$SUREFIRE_RERUN_FAILING_COUNT -nsu -B -Pauth-server-quarkus,auth-server-fips140-2,app-server-wildfly -Dcom.redhat.fips=false -Dredhat.crypto-policies=false $STRICT_OPTIONS -Dtest=$TESTS -pl testsuite/integration-arquillian/tests/base 2>&1 | misc/log/trimmer.sh
 if [ $? -ne 0 ]; then
   exit 1
 fi
 
 # New Base Tests
-./mvnw package -nsu -B -Dredhat.crypto-policies=false -Dtest=$TESTSUITE_NAME -pl tests/base
+./mvnw package -nsu -B -Dcom.redhat.fips=false -Dredhat.crypto-policies=false -Dtest=$TESTSUITE_NAME -pl tests/base

--- a/.github/scripts/run-fips-ut.sh
+++ b/.github/scripts/run-fips-ut.sh
@@ -1,20 +1,22 @@
 #!/usr/bin/env bash
 
+JAVA_VERSION=21
+
 rm -f /etc/system-fips
-dnf install -y java-25-openjdk-devel crypto-policies-scripts
+dnf install -y "java-${JAVA_VERSION}-openjdk-devel"
 fips-mode-setup --enable --no-bootcfg
 fips-mode-setup --check | grep "FIPS mode is enabled."
 if [ $? -ne 0 ]; then
   exit 1
 fi
-export JAVA_HOME=/etc/alternatives/java_sdk_25
+export JAVA_HOME="/etc/alternatives/java_sdk_${JAVA_VERSION}"
 
 # Build all dependent modules
 ./mvnw install -nsu -B -am -pl crypto/default,crypto/fips1402 -DskipTests
 
-./mvnw test -nsu -B -pl crypto/default,crypto/fips1402 -Dredhat.crypto-policies=true
+./mvnw test -nsu -B -pl crypto/default,crypto/fips1402 -Dcom.redhat.fips=true -Dredhat.crypto-policies=true
 if [ $? -ne 0 ]; then
   exit 1
 fi
 
-./mvnw test -nsu -B -pl crypto/default,crypto/fips1402 -Dredhat.crypto-policies=true -Dorg.bouncycastle.fips.approved_only=true
+./mvnw test -nsu -B -pl crypto/default,crypto/fips1402 -Dcom.redhat.fips=true -Dredhat.crypto-policies=true -Dorg.bouncycastle.fips.approved_only=true

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionResetPasswordTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionResetPasswordTest.java
@@ -45,7 +45,6 @@ import org.keycloak.services.resources.LoginActionsService;
 import org.keycloak.testframework.events.EventAssertion;
 import org.keycloak.testframework.realm.UserBuilder;
 import org.keycloak.testsuite.ActionURIUtils;
-import org.keycloak.testsuite.AssertEvents;
 import org.keycloak.testsuite.admin.AdminApiUtil;
 import org.keycloak.testsuite.pages.ErrorPage;
 import org.keycloak.testsuite.pages.LoginConfigTotpPage;
@@ -515,14 +514,14 @@ public class AppInitiatedActionResetPasswordTest extends AbstractAppInitiatedAct
                         + "auth_session_id=" + authSessionId))) {
                     Assert.assertEquals(302, response2.getStatusLine().getStatusCode());
                 }
-                events.expect(EventType.RESTART_AUTHENTICATION).user((String) null).assertEvent();
+                EventAssertion.assertSuccess(events.poll()).type(EventType.RESTART_AUTHENTICATION).userId(null);
 
                 // authenticate the user in the browser
                 oauth.openLoginForm();
                 loginPage.assertCurrent();
                 loginPage.login("test-user@localhost", "password");
                 appPage.assertCurrent();
-                events.expectLogin().user(AssertEvents.isUUID()).detail(Details.USERNAME, "test-user@localhost").assertEvent();
+                EventAssertion.expectLoginSuccess(events.poll()).hasUserId().details(Details.USERNAME, "test-user@localhost");
 
                 // navigate to the authenticate page with the other auth_session_id, tab_id and client_data
                 driver.navigate().to(oauth.getBaseUrl() + "/realms/" + oauth.getRealm()
@@ -531,8 +530,8 @@ public class AppInitiatedActionResetPasswordTest extends AbstractAppInitiatedAct
                         + "&tab_id=" + attrs.get("tab_id")
                         + "&client_data=" + attrs.get("client_data"));
                 errorPage.assertCurrent();
-                events.expect(EventType.LOGIN_ERROR).error(Errors.INVALID_CODE).user((String) null)
-                        .detail(Details.REASON, "cookie does not match auth_session query parameter").assertEvent();
+                EventAssertion.assertError(events.poll()).type(EventType.LOGIN_ERROR).error(Errors.INVALID_CODE).userId(null)
+                        .details(Details.REASON, "cookie does not match auth_session query parameter");
             }
         }
     }

--- a/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/UserInfoTest.java
+++ b/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/UserInfoTest.java
@@ -745,13 +745,12 @@ public class UserInfoTest extends AbstractKeycloakTest {
 
             assertEquals(Status.UNAUTHORIZED.getStatusCode(), response.getStatus());
 
-            events.expect(EventType.USER_INFO_REQUEST_ERROR)
+            EventAssertion.assertError(events.poll()).type(EventType.USER_INFO_REQUEST_ERROR)
                     .error(Errors.INVALID_TOKEN)
-                    .user(Matchers.nullValue(String.class))
-                    .session(Matchers.nullValue(String.class))
-                    .detail(Details.AUTH_METHOD, Details.VALIDATE_ACCESS_TOKEN)
-                    .client((String) null)
-                    .assertEvent();
+                    .userId(null)
+                    .sessionId(null)
+                    .details(Details.AUTH_METHOD, Details.VALIDATE_ACCESS_TOKEN)
+                    .clientId(null);
 
             rep.setNotBefore(0);
             realm.update(rep);

